aten/src/ATen/native/TensorShape.cpp

```diff
@@ -42,6 +42,50 @@ Tensor& set_(Tensor& result, Storage source) {
   return result.set_(source, 0, static_cast<int64_t>(source.size()), {});
 }
 
+// unify with cuda implementation?  This is not done to avoid a dispatch in resize_impl_cpu_
```

Comment:
It looks like big stretches of the error checking etc. are identical between these two - would it be a hassle to factor that stuff out into a common helper?

Reply:
I think that would work, let me try.